### DISCUSSION TITLE
Suggestion for AUX scanner documentation

### DISCUSSION THREAD

**Discussion by joelnitta** [retorquere/zotero-better-bibtex#1918] [General]
It is great that the AUX scanner makes it easy to automatically generate a collection in zotero. Before that feature was available, to make a zotero collection from a set of keys, I had to manually click the "+" button in the advanced search and type a key each time... ugh!

However, I think the [documentation on the website](https://retorque.re/zotero-better-bibtex/citing/aux-scanner/) could be easier to understand, especially for BBT users that don't actually use latex directly (yes we exist! in my case thanks to Rmarkdown that pleasantly hides all of that away).

For the purposes of generating a collection from an AUX file, all the AUX file really needs to include is this:

```
\citation{citekey1}
\citation{citekey2}
...
```

I was hitting my head against a wall trying to figure out how to make an AUX file until I realized this.

If such an example were provided, it would probably make this awesome feature easier for others to use.

**Comment by retorquere** [comment]
I'm sort of blind to these deficiencies in the docs as I don't know what others won't know. The pages have an "edit this page" in the top-right corner - you can modify as you please, and I can merge those changes into the site.

**Reply by joelnitta** [reply]
Thanks, I'll submit a PR with some suggested changes when I get a chance.

BTW I figured out what was needed in the AUX file from [your example in this issue](https://github.com/retorquere/zotero-better-bibtex/issues/1503#issuecomment-618430041)... so even just an example file like that in the documentation would have been helpful (for me anways).

**Reply by retorquere** [reply]
But the aux file is just a generated artifact from the build process. Why would you need to know what is in it?

**Reply by joelnitta** [reply]
My workflow is perhaps a bit different from what you were expecting, I think it will help if I explain it: 

I write in Rmarkdown. I copy citations from zotero with BBT citation key quick copy, and insert them into the Rmd file as I go. I need to have a bib (actually I use YAML, but you get the idea) file with just the references I cite in my paper, but since it is a pain to manually curate a reference group, I export the entire library from zotero (on auto-mode so it's always up to date). I have an R script that scans my Rmd file, extracts all the of the keys, filters the exported library, and saves it as the custom reference file for that manuscript.

This works fine, but often I need to go back and edit the entries in Zotero to fix some details that didn't get imported correctly. It is nice to be able to browse the list of references I have currently cited in a particular paper rather than search each one down individually. So that is where the "scan aux file" is a great feature. But it my case I don't use Latex to render the Rmd (well it's involved somewhere in there with pandoc, but it's pretty well buried and hard to get at the build artifacts). So once I again I use my script to scan my Rmd file and produce the list of citation keys. I need to export them **as if** they are an AUX file. So that's why I needed to the know the structure of the file.

**Reply by retorquere** [reply]
Alright.

**Comment by joelnitta** [comment]
Follow up: sorry, haven't filed that PR yet, but I just noticed that when I added some references this way, the matching seemed to include partial strings. For example, "Park2020" in the AUX list added "Park2020a" as well as "Park2020" in the Zotero references

**Reply by retorquere** [reply]
Do you have a sample for this? This shouldn't be the case. I search for exact matches (case-sensitive even), so it must be that the collection of the citation keys from the aux file goes wrong.

**Reply by joelnitta** [reply]
Sorry for the late reply... just tried it again with much shorter AUX file (one entry, either "Park2020" or "Park2020a") and it worked correctly, didn't add both in the Zotero references. So it must have been something else on my end.

**Reply by retorquere** [reply]
If you ever do see it happen again you can just reopen the issue.
